catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/framework/FrameworkUtilities.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.framework;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import org.codice.alliance.video.stream.mpegts.rollover.RolloverActionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+
+public class FrameworkUtilities {
```

Comment:
I'll change the core of the code to failsafe. But there's an initial sleep that doesn't fit within the failsafe model. I'll also rename the class. And I'll make it non-static.